runtime/executor/src/main/java/edu/snu/onyx/runtime/executor/data/blocktransfer/BlockInputStream.java

```diff
@@ -112,7 +110,11 @@ void append(final ByteBuf byteBuf) {
    */
   void markAsEnded() {
     byteBufInputStream.byteBufQueue.close();
-    completeFuture.complete(inputStreamIterator);
+    try {
+      completeFuture.complete(new DataUtil.InputStreamIterator<>(byteBufInputStream, serializer));
+    } catch (final IOException e) {
+      throw new RuntimeException(e);
```

Comment:
Please be careful to throw `RuntimeException`.
Throwing `IOException` explicitly is much better for debugability.